libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/internal/definitions.kt

```diff
@@ -0,0 +1,308 @@
+package com.apollographql.apollo3.ast.internal
+
+/**
+ * This file contains several groups of GraphQL definitions we use during codegen:
+ *
+ * - builtins.graphqls: the official built-in defintions such as [built-in scalars](https://spec.graphql.org/draft/#sec-Scalars.Built-in-Scalars), [built-in directives](https://spec.graphql.org/draft/#sec-Type-System.Directives.Built-in-Directives) or [introspection definitions](https://spec.graphql.org/draft/#sec-Schema-Introspection.Schema-Introspection-Schema).
+ * - link.graphqls: the [core schemas](https://specs.apollo.dev/link/v1.0/) definitions.
+ * - apollo-${version}.graphqls: the client directives supported by Apollo Kotlin. Changes are versioned at https://github.com/apollographql/specs. Changing them requires a new version and a PR.
+ *
+ * TODO: move to apollo-compiler instead?
```

Comment:
My 2c: since it's used only in AST, it makes sense to keep it here.

Reply:
I'd like `apollo-ast` to be as general and as small as possible. I'd say apollo specific validation do not belong there. It could even be argued if validation belongs there at all actually since android apps will not need it (because the operations were validated at build time already). But let's see how it unfolds....

Reply:
> I'd say apollo specific validation do not belong there

Yeah - actually that makes a lot of sense!